### PR TITLE
Add support for indirect translation function calls in JS

### PR DESCRIPTION
**Pull request by Tug**
This change allows makepot on JS files to work for babel transpiled files. This change is similar to https://github.com/wp-cli/i18n-command/pull/204 but related to a babel behavior instead.

In [some cases](https://github.com/babel/babel/blob/91fb64eaf53c67a7ed7605aab71f8dcaa4d39ae2/packages/babel/src/transformation/modules/lib/remaps.js#L32), babel will convert simple function calls to indirect ones (in the form of `(0, __)( 'translate' )`).

I guess this is done so that legacy ES5 script that uses strict mode are able retrieve the global `window` object using `this`.

In any case, babel often transpiles code like:
```js
import { __ } from '@wordpress/i18n';

__( 'translate' );
```

Into:

```js
var _i18n = _$$_REQUIRE(_dependencyMap[4], "@wordpress/i18n");

(0, _i18n.__)( 'translate' );
```

### Testing instructions
- Download this JS file which uses `@wordpress/i18n` and is transpiled by babel (bundled by metro.js): https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/bundle/android/App.js
- Run `wp i18n make-pot . --include=App.js -ignore-domain gutenberg-android.pot`
- Notice strings are correctly extracted with this patch, otherwise the POT file is empty

### CODE REVIEW

**Comment by swissspidy** [comment]
Really nice, thank you!

File: features/makepot.feature [hunk 1]
```diff
@@ -1615,6 +1615,11 @@ Feature: Generate a POT file of a WordPress project
 
       Object(u.__)( 'minified.__', 'foo-plugin' );
       Object(j._x)( 'minified._x', 'minified._x_context', 'foo-plugin' );
+
+      /* translators: babel */
```
Comment (swissspidy) [inline features/makepot.feature:1619]: Nit: could test whether the comment is extracted successfully

Comment (Tug) [reply features/makepot.feature:1619]: Thanks, just added that 👍

**Review by swissspidy — APPROVED** [review]

**Comment by Tug** [comment]
Thanks for the review @swissspidy 

I just added a last minute check to exclude a edge case: `(0, __, somethingElse)( 'translate' )`

**Comment by swissspidy** [comment]
No idea why the tests are timing out 😕

**Comment by Tug** [comment]
Just force pushed to give Travis another chance.

**Comment by Tug** [comment]
Seems like the [test jobs for ubuntu xenial with php 5.6](https://travis-ci.org/github/wp-cli/i18n-command/builds/706207597) are already close to 10min on the master branch, so it's possible we're reaching travis limit just by adding those few tests. Just experimenting with travis config a bit here ^

**Comment by swissspidy** [comment]
@schlessera any thoughts here?

**Comment by Tug** [comment]
It's green now, and it looked like those tests indeed needed a couple more minutes to finish. Happy to leave https://github.com/wp-cli/i18n-command/pull/228/commits/5b9d70e1ccc046030fa741873f76e76e6d7ed897 in if you think it's the right approach

**Comment by swissspidy** [comment]
Interesting! Thanks for debugging.

I _think_ for the sake of consistency with all other repos we should undo the `travis_wait` part again. But maybe Alain has some thoughts here.

**Review by schlessera — CHANGES_REQUESTED** [review]
As @swissspidy already mentioned, the `travis.yml` file is supposed to be shared across all repos, so I'd like to find a solution that doesn't require changes to that.

The problem seems to be with PHP 5.6 only, so I assume there's something being used that has an extension on PHP 7+ but not on PHP 5.6.

So we'd need to find out what it is that is so much slower.
If there's an extension we can install to accelerate PHP 5.6 builds, I'm happy to add it centrally to all repos.
If not, we can skip the most expensive tests by strategically marking some tests as `@require-php-7-0`.

**Comment by schlessera** [comment]
Hmm, if Travis times out after 10 minutes on Behat, this means Behat didn't produce a single `.` for more than 10 minutes...

So there seems to be a single operation in the tests somewhere that runs for a very long time...

File: src/JsFunctionsScanner.php [hunk 2]
```diff
@@ -247,6 +247,36 @@ private function resolveExpressionCallee( Node\CallExpression $node ) {
...
 		}
 
+		// If the callee is an indirect function call as created by babel, resolve it.
+		// For example: (0, u.__)( "translation" ).
+		if (
+			'ParenthesizedExpression' === $callee->getType() &&
+			'SequenceExpression' === $callee->getExpression()->getType() &&
+			2 === count( $callee->getExpression()->getExpressions() ) &&
+			'Literal' === $callee->getExpression()->getExpressions()[0]->getType() &&
+			[] !== $node->getArguments()
```
[the remark this answers is not in the document]
Comment (Tug) [reply src/JsFunctionsScanner.php:257]: This file still has the operators at the end of the line, not sure if you'd like me to update the whole file or just this condition?

[the remark this answers is not in the document]
Comment (schlessera) [reply src/JsFunctionsScanner.php]: Oh, then nevermind, let's stick to consistency for now then.

**Comment by schlessera** [comment]
I'll try to add a speed listener of some sort to Behat, reporting on the slow(est) scenarios and/or steps.

**Comment by simison** [comment]
Hi folks! Anything in particular stopping this PR from proceeding?

**Comment by swissspidy** [comment]
> Hi folks! Anything in particular stopping this PR from proceeding?

I think mainly the issue with the tests timeout needs to be addressed.

**Review by schlessera — APPROVED** [review]
